websites/A/AnimeGO/presence.ts

```diff
@@ -0,0 +1,142 @@
+const presence = new Presence({
+	clientId: "935597176426491924"
+});
+
+async function getStrings() {
+	return presence.getStrings({
+		play: "general.playing",
+		pause: "general.paused"
+	});
+}
+
+let strings: Awaited<ReturnType<typeof getStrings>>,
+	video = {
+		duration: 0,
+		currentTime: 0,
+		paused: true
+	};
+
+presence.on(
+	"iFrameData",
+	(data: { duration: number; currentTime: number; paused: boolean }) => {
+		video = data;
+	}
+);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		details: "Где-то на сайте",
+		largeImageKey: "animego_logo",
+		smallImageText: "AnimeGO"
+	};
+	if (!strings) strings = await getStrings();
+
+	const typeContent = document
+			.querySelector("meta[property='og:url']")
+			.getAttribute("content")
+			.split("/")[3],
+		typeCurrent =
+			typeContent === "anime"
+				? "аниме"
+				: typeContent === "manga"
+				? "манги"
+				: typeContent === "character" || typeContent === "characters"
+				? "персонажа"
+				: "человека";
+
+	if (document.location.pathname === "/")
+		presenceData.details = "На главной странице";
+	else if (document.location.pathname.includes("/profile")) {
+		presenceData.details = `Смотрит свой профиль (${
+			document.querySelector(".nav.navbar-nav.login li:nth-child(1) span")
+				.textContent
+		})`;
+	} else if (document.location.pathname.includes("/user/")) {
+		presenceData.details = `Смотрит профиль ${
+			document
+				.querySelector("meta[property='og:url']")
+				.getAttribute("content")
+				.split("/")[4]
+		}`;
+		if (
+			document.location.pathname.includes("/anime") ||
+			document.location.pathname.includes("/manga")
+		)
+			presenceData.state = document.querySelector("main h1").textContent;
+	} else if (
+		document.location.pathname === "/anime" ||
+		document.location.pathname.includes("/anime/season/") ||
+		document.location.pathname.includes("/anime/genre/") ||
+		document.location.pathname.includes("/anime/type/") ||
+		document.location.pathname.includes("/anime/status/") ||
+		document.location.pathname.includes("/anime/filter/") ||
+		document.location.pathname.includes("/anime/studio/") ||
+		document.location.pathname === "/manga" ||
+		document.location.pathname.includes("/manga/season/") ||
+		document.location.pathname.includes("/manga/genre/") ||
+		document.location.pathname.includes("/manga/type/") ||
+		document.location.pathname.includes("/manga/status/") ||
+		document.location.pathname.includes("/manga/filter/") ||
```

Comment:
can't you just check if it includes /manga?

Reply:
`<String>.startsWith("/anime")`
`<String>.startsWith("/manga")`

Reply:
> can't you just check if it includes /manga?

In this case, links to anime/manga will not be displayed correctly. On the site, links to anime look something like this: `https://animego.org/anime/gospozha-kaguya-v-lyubvi-kak-na-voyne-3-1981`, that is, after `/anime/`. Links to sections with search by genre, type, etc. look almost the same, which are written separately.

Reply:
In that case you should make a single regex that accounts for all of these